libraries/shared/factories/storage/keys_lookup.go

```diff
@@ -68,11 +69,7 @@ func (lookup *keysLookup) Lookup(key common.Hash) (types.ValueMetadata, error) {
 func (lookup *keysLookup) refreshMappings() error {
 	var err error
 	lookup.mappings, err = lookup.loader.LoadMappings()
-	if err != nil {
-		return err
-	}
-	lookup.mappings = storage.AddHashedKeys(lookup.mappings)
-	return nil
+	return err
```

Comment:
Depending on the implementation of LoadMappings it's possible that you'll overwrite lookup.mappings on an error. Now maybe that's the desired behavior, I'm not sure, but the other option would be to check the error code before you update the mappings data.

Reply:
Good point. I don't really have a strong opinion either way - if there's an error, it needs to be returned and we need to retry. If the mappings aren't refreshed successfully, then we won't be able to recognize the storage key - regardless of whether `lookups.mappings` is overwritten. That said, it seems like good practice to avoid assigning instance variables when there's an error.